jobtracker.go

```diff
@@ -250,112 +217,179 @@ func (jt *jobTracker) finishedJobLoop() {
 
 		// This sorts in reverse order of start time so that we'll fill in the
 		// details for the newest jobs first.
-		sort.Sort(sort.Reverse(jobDetails(finished.Jobs.Job)))
-		log.Println("Backfill jobs:", len(finished.Jobs.Job))
-		for i := range finished.Jobs.Job {
+		sort.Sort(sort.Reverse(jobDetails(backfill.Jobs.Job)))
+		total := len(backfill.Jobs.Job)
+		log.Println("Jobs to backfill:", total)
+		for i := range backfill.Jobs.Job {
 			if i > jobLimit {
 				break
 			}
-			jt.finished <- &job{Details: &finished.Jobs.Job[i], host: jt.hs}
+
+			if i%100 == 0 {
+				log.Printf("Backfilled %d/%d jobs", i, total)
+			}
+
+			jt.backfill <- &job{Details: backfill.Jobs.Job[i], running: false}
 		}
+
+		log.Println("Finished backfilling jobs.")
 	}()
 
 	for _ = range time.Tick(*pollInterval) {
-		finished := &jobResp{}
-		if err := jt.ListJobs(jt.hs, finished); err != nil {
-			log.Println("Error listing finished jobs:", err)
+		dur := 1 * time.Minute
+		if dur < (*pollInterval * 2) {
+			dur = *pollInterval * 2
+		}
```

Comment:
pollInterval is configurable - this makes it so that if the pollInterval is set to say, 5 minutes, we don't load only load jobs for the last minute.